src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let join = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (
+        this[i] !== null
+        && typeof this[i] !== 'undefined'
+      ) {
```

Comment:
you can simplify without using `typeof`
```suggestion
      if (this[i] !== null && this[i] !== undefined) {
```